hummingbot/model/decimal_type_decorator.py

```diff
@@ -0,0 +1,30 @@
+from decimal import Decimal
+
+from sqlalchemy import BigInteger, TypeDecorator
+
+
+class SqliteDecimal(TypeDecorator):
+    # This TypeDecorator use Sqlalchemy BigInteger as impl. It converts Decimals
+    # from Python to Integers which is later stored in Sqlite database.
+    impl = BigInteger
+
+    def __init__(self, scale):
+        # It takes a 'scale' parameter, which specifies the number of digits
+        # to the right of the decimal point of the number in the column.
+        TypeDecorator.__init__(self)
+        self.scale = scale
+        self.multiplier_int = 10 ** self.scale
+
+    def process_bind_param(self, value, dialect):
+        # e.g. value = Column(SqliteDecimal(2)) means a value such as
+        # Decimal('12.34') will be converted to 1234 in Sqlite
+        if value is not None:
+            value = int(Decimal(value) * self.multiplier_int)
+        return value
+
+    def process_result_value(self, value, dialect):
+        # e.g. Integer 1234 in Sqlite will be converted to Decimal('12.34'),
+        # when query takes place.
+        if value is not None:
+            value = Decimal(value) / self.multiplier_int
+        return value
```

Comment:
I think the comments here should be docstrings instead

Reply:
Done